### ISSUE TITLE
Proxy settings for skyHttpClient are lost when tls is enabled

### ISSUE THREAD

**Issue by asehra** [concourse/concourse#8296]
Our Credentials provider sits behind a proxy, and if the TLS is enabled on concourse, then the HTTP Client that speaks to the credentials provider, fails to connect. This is because the `HTTPTransport` used when creating the client, replaces the default transport while enabling TLS [here](https://github.com/concourse/concourse/blob/master/atc/atccmd/command.go#L1416).

**Comment by xtremerui** [comment]
Could you give an example of the credentials provider TLS configuration and the errors in ATC log?

**Comment by asehra** [comment]
I am using AWS SecretsManager as the credentials provider, however in my setup, I must access it via a corporate proxy (don't ask why 🤦). With "if TLS is enabled", I mean the TLS on the concourse web process being enabled.  I configured the web process to use TLS  using a self-signed cert, and bound it to port 443 as usual. 

When ATC tries to fetch the credentials from the secrets manager, the connection hangs for a while until it fails with this error log (expanded for readability):
```json
{
  "secretsmanager": {
    "aws_region":"us-east-1",
    "health": {
      "error":"RequestError: send request failed\ncaused by: Post \"[https://secretsmanager.us-east-1.amazonaws.com/\":](https://secretsmanager.us-east-1.amazonaws.com//%22:) dial tcp 18.214.158.206:443: connect: connection timed out",
      "method":"GetSecretValue"
    },
    "pipeline_secret_template":"/concourse/{{.Team}}/{{.Pipeline}}/{{.Secret}}",
    "shared_secret_template":"/concourse/{{.Secret}}",
    "team_secret_template":"/concourse/{{.Team}}/{{.Secret}}"
  }
}
```

When we looked at the concourse source code that we [linked to](https://github.com/concourse/concourse/blob/master/atc/atccmd/command.go#L1416) in the description, it gave us a hunch that the if condition in there replaces the default settings in `DefaultTransport` (including ProxyFromEnvironment) when it builds a new `&http.Transport{` object. Which we thought would mean that if the concourse web process is started with TLS enabled, it is not going to use the Proxy from environment variables in this http client which would fetch the secrets from secrets manager. 

So we tried launching our concourse without TLS enabled to prove the point, and as guessed, the concourse was able to fetch the secrets from the secretsmanager, because it used the proxy from environment variables.
